### DISCUSSION TITLE
Prune points older than N days

### DISCUSSION THREAD

**Discussion by oransterf** [Freika/dawarich#336] [Q&A] [answered]
Hello, thank you for developing this software, it's really useful!

I had a question about the number of points and data storage requirements.
Right now I have ~5000 points and the db_data is about ~80MB. 
I wanted to know if there's any compression possible or if it will grow linearly at this rate?

Relatedly, is there a way to set a maximum number of points or of days to keep points in the database to prune older points?

Finally, how many points are too many points? I am using this with a device in my car so it is always powered when the car is powered on, and I currently have it set to about 1 point per 15 seconds.  Is this a lot? What scale is the app made to support?

Thanks again!

**Comment by Freika** [comment, marked as answer]
There is no pruning or compression functionality currently, and there are no plans to introduce either.

The database might get bigger and bigger due to the way how reverse geocoding information is being stored separately for each and every point. The size growth will probably not be linear as reverse geocoding info is sometimes bigger than the point data itself.

It's up to you to decide how often your points are being recorded, my config is set up to record every 10 seconds or 50 meters, depending on what happens sooner. Works well for me both as a pedestrian and as a driver. A few days worth of driving usually adds about 800-2500 points to the database.

Some people have millions of points imported from their Google Takeout, I have about 700k.

The main problem for now is the browser performance on the Map page, and it depends on the hardware used. My Macbook M1Pro starts struggling at around 50k points being rendered on the map. I hope to improve this performance in the future to support 100k+ points being rendered.

**Reply by oransterf** [reply]
Thank you for answering, that makes sense!
